stdpopsim/slim_engine.py

```diff
@@ -1833,6 +1839,41 @@ def _get_msp_rate_map(breaks, is_this_dfe, rate):
                 )
             )
 
+        # If haploid, rebuild individual table so each sample is an individual
+        if contig.ploidy == 1:
+
+            def to_ragged(x, idx):
+                return [x[idx[i] : idx[i + 1]] for i in range(len(idx) - 1)]
+
+            tables = ts.dump_tables()
+            indiv = tables.individuals.asdict()
+            indiv["flags"] = np.repeat(indiv["flags"], 2)
+            for field in ["location", "parents", "metadata"]:
+                offset = indiv[field + "_offset"]
+                ragged = to_ragged(indiv[field], offset)
+                indiv[field] = np.array([x for x in ragged for _ in range(2)]).flatten()
+                offset_length = np.insert(np.repeat(np.diff(offset), 2), 0, 0)
+                indiv[field + "_offset"] = np.cumsum(offset_length)
+
+            tables.individuals.set_columns(
+                flags=indiv["flags"],
+                location=indiv["location"],
+                location_offset=indiv["location_offset"],
+                parents=indiv["parents"],
+                parents_offset=indiv["parents_offset"],
+                metadata=indiv["metadata"],
+                metadata_offset=indiv["metadata_offset"],
+                metadata_schema=indiv["metadata_schema"],
+            )
+
+            samples = tables.nodes.flags == tskit.NODE_IS_SAMPLE
```

Comment:
In case there are other flags in the future, it'd be better to do a bitwise or here. (I think this is
```
samples = ( tables.nodes.flags | tskit.NODE_IS_SAMPLE > 0)
```
... ?